### PR TITLE
Add speed test

### PR DESCRIPTION
**Pull request by jessicaaustin**
Addresses https://github.com/ioos/ioos_qc/issues/36

Changes:

* Add new module `argo`
  * this module defines tests based on ARGO QC Manual
* Deprecated `glider` module and moved `glider.pressure_check` to `argo.pressure_increasing_test`
* add `argo.speed_test`
* use pygc instead of pyproj for dist calc
  * this uses Karney instead of Vincenty for distance calculation. Karney converges for all points, whereas Vincenty fails to converge for nearly antipodal points
  * this affects existing test `qartod.location_test` when `range_max` is used
* moving a few things to `utils` so they can be shared between `qartod` and `moving_platforms` -- nothing changed during the move

Questions/discussion for reviewers:

* ~~Thoughts on the `moving_platforms` module? is the name ok? any desire to group with `gliders`?~~ -- resolved, see comments below

### CODE REVIEW

**Comment by jessicaaustin** [comment]
@kwilcox could you please review?

@lukecampbell if you have a chance, could you please check the distance calcs specifically?

Thanks!

File: ioos_qc/moving_platforms.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        flag_arr[speed > suspect_threshold] = QartodFlags.SUSPECT
+
+    with np.errstate(invalid='ignore'):
+        flag_arr[speed > fail_threshold] = QartodFlags.FAIL
+
+    # first value is unknown, since we have no speed data for the first point
+    flag_arr[0] = QartodFlags.UNKNOWN
+
+    # If the value is masked set the flag to MISSING
+    flag_arr[lon.mask] = QartodFlags.MISSING
```
Comment (kwilcox) [inline ioos_qc/moving_platforms.py:102]: Should this be `flag_arr[dist.mask]`, the `lon.mask` is already checked together with the `lat.mask` and the mask on distance is never checked.

**Review by kwilcox — CHANGES_REQUESTED** [review]

**Comment by kwilcox** [comment]
`moving_platforms` a very Axiom-heavy term... but does describe this well. We could start with a `platforms` folder and nest different types there (`ioos_qc/platforms/moving`)? I don't feel strongly, we can leave it like this and see how it grows.

**Comment by jessicaaustin** [comment]
Yeah I was trying to think of something generic; "moving platforms" sounded generic enough to me? `¯\_(ツ)_/¯` It could describe gliders, AUVs, floats, animals, etc.

We could also call it ARGO -- the [ARGO QC manual](https://github.com/ioos/ioos_qc/files/5404792/argo-quality-control-manual.pdf) describes both the speed test and the pressure check (both on pg 7). That would fit the pattern of naming modules after manuals.

Or we could just have a "general" module so we don't have to worry so much about what bucket a test falls into.

File: ioos_qc/moving_platforms.py (hunk 1 above)
Comment (jessicaaustin) [reply ioos_qc/moving_platforms.py]: ty, fixed

**Comment by kwilcox** [comment]
Let's go with either `argo` or `general` and put the pressure check there as well. Your call.

**Comment by jessicaaustin** [comment]
Ok I'm gonna do `argo`. That's what my heart says

**Comment by jessicaaustin** [comment]
Ok @kwilcox i'm happy with merging this if you are

**Review by kwilcox — APPROVED** [review]

**Pull request merged**